mava/systems/tf/mappo/training.py

```diff
@@ -414,32 +507,42 @@ def _backward(self) -> None:
             agent_key = self._agent_net_keys[agent]
 
             # Get trainable variables.
-            policy_variables = (
-                self._observation_networks[agent_key].trainable_variables
-                + self._policy_networks[agent_key].trainable_variables
-            )
-            critic_variables = self._critic_networks[agent_key].trainable_variables
+            if self._use_single_optimizer:
+                policy_variables = (
+                    self._policy_networks[agent_key].trainable_variables
+                    + self._critic_networks[agent_key].trainable_variables
+                    + self._observation_networks[agent_key].trainable_variables
+                )
+            else:
+                policy_variables = self._policy_networks[agent_key].trainable_variables
+                critic_variables = (
+                    self._critic_networks[agent_key].trainable_variables
+                    + self._observation_networks[agent_key].trainable_variables
```

Comment:
When using two optimisers should we not have the obs network parameters be updated with the policy network? I think this is how we had it before, but open to hearing a possible motivation for the change.

Reply:
Discussed in point above. (Summary acme only have obs network vars updated by critic optim/network, possibly because of critic multiplexer).